File: PythonImplementations/Problem03.py
# Problem 3: Largest Prime Factor
# find the largest prime factor of a number
# if N is not specified, 600851475143 is used as N
# usage:
# $ python Problem03.py [N = 600851475143]
from PEUtils import PrimeFactors
import sys

# use the PrimeFactors function defined in PEUtils
def main():
    N = int(sys.argv[1]) if len(sys.argv) > 1 else 600851475143
    print max(PrimeFactors(N))

main()

File: RustImplementations/Problem03.rs
/*
  Problem 3: Largest Prime Factor
  Find the largest prime factor of the number 600851475143
*/
extern crate PEUtils;
use PEUtils::PrimeNumbers;

fn main() {

    for p in PrimeNumbers::new()
        .take_while(|x| *x < 7920) {
        println!("{}", p);
    }
    /*
    let max_prime_factor = PrimeFactors(600851475143)
        .fold(0, |b, a| max(b, a))
    println!("{}", max_prime_factor);
    */
}
